RadixWallet/Features/Signing/Children/SignWithFactorSource.swift

```diff
@@ -24,7 +24,8 @@ public struct SignWithFactorSource: Sendable, FeatureReducer {
 				self.factorSourceAccess = .init(kind: .device, purpose: .signature)
 			case .ledger:
 				assert(signingFactors.allSatisfy { $0.factorSource.kind == LedgerHardwareWalletFactorSource.kind })
-				self.factorSourceAccess = .init(kind: .ledger(nil), purpose: .signature)
+				let ledger: LedgerHardwareWalletFactorSource? = signingFactors.first?.factorSource.extract()
```

Comment:
Hmm this looks wrong? There is a missing filter and or `first(where { $0.kind == .ledger})`?

Reply:
if you look on the above line, we are asserting that every signing factor meets such condition

Reply:
maybe im sicker than I thought :)